interview-questions.md

```diff
@@ -7,48 +7,55 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is an instance variable in Ruby? How is it different from other variables?
 
-  Your answer:
+  Your answer: The syntax for an instance variable is @ followed by the variables assigned name. This is (as the name reveals) is an instance of the chosen variable and not a direct call of the variable itself. 
 
-  Researched answer:
+  Researched answer: After digging a little deeper I learned that what follows the instance variable while its being assigned is actually it's scope. Also i discovered that instance variables are not global.
 
 
 
 2. What is a block in Ruby?
 
-  Your answer:
+  Your answer: Blocks in ruby are built in methods that require a do/end
 
-  Researched answer:
+  Researched answer: After researching blocks a little more I discovered that blocks are anonymous pieces of code that can also use curly braces {} instead of do/end
 
 
 
 3. Ruby has an implicit return. What does that mean?
 
-  Your answer:
+  Your answer: This means that the last line of code that you write in a program will automatically be attempted to be returned by Ruby
 
-  Researched answer:
+  Researched answer: To improve on my previous definition, I would add that an implict return is automatically called upon the last executed instruction in a method when an explicit return has not been used. 
 
 
 
 4. What is object-oriented programming? How is it different than functional programming?
 
-  Your answer:
+  Your answer: This is a programming mind set that focuses on the functionality of objects and thus builds around that functionailty 
```

Comment:
Lets pay close attention to these questions, its asking two things- One, what is OOP? then, it asks "How is it different from functional programming?" 

Your answer only scopes a bit on OOP, so lets be mindful of that with our answers.